test/run_test.py

```diff
@@ -584,6 +585,8 @@ def test_distributed(test_module, test_directory, options):
                     return return_code
             finally:
                 shutil.rmtree(tmp_dir)
+                os.environ.clear()
+                os.environ.update(old_environ)
```

Comment:
@jeffdaily  Could this also be why the sharding for ROCm distributed caused issues?

Reply:
It looks like this resolves the issue with the rocm sharded dist tests, as well.